### PR TITLE
Stargate white list

### PR DESCRIPTION
**Pull request by ThanhNhann**
Closes: #2627 

## What is the purpose of the change

Extract the query path and the correct proto struct from the query.pb.go files so we don't need to add Stargate Querier Whitelist by hard coded 

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (no)
  - How is the feature or change documented? (not documented)

### CODE REVIEW

**Comment by ValarDragon** [comment]
This doesn't seem right, looks like its adding every query per module?

Ideally we could just give it a copy of the struct, and then it can extract the route from that

**Comment by mattverse** [comment]
yeah, we should not be registering all the queries for the modules, but only add the queries that we've selected.

And I also agree with @ValarDragon  on the point that ideally, we solve this in the way we give a list of structs we want to add  and we have an automated part or logic that routes with with it's paths

**Comment by mattverse** [comment]
> How about the list of struct you say @mattverse, I think it is like query proto struct or request struct right? If not, can you give me more information about this?

Yes it is!

**Comment by ValarDragon** [comment]
Nice!

**Comment by hieuvubk** [comment]
> Nice!

Thank you! It's would be great if we implement it

File: wasmbinding/stargate_whitelist.go [hunk 1]
```diff
@@ -32,92 +36,122 @@ import (
 // thread safe sync.Map.
 var stargateWhitelist sync.Map
 
+// This is
```
Comment (mattverse) [inline wasmbinding/stargate_whitelist.go:39]:
```suggestion
```

**Review by mattverse — COMMENTED** [review]
Added some initial comments and reviews for this! Please take a look!

File: wasmbinding/stargate_whitelist.go [hunk 2]
```diff
@@ -32,92 +36,122 @@ import (
 // thread safe sync.Map.
 var stargateWhitelist sync.Map
```
[the remark this answers is not in the document]
Comment (hieuvubk) [reply wasmbinding/stargate_whitelist.go:38]: Yes It is

File: wasmbinding/stargate_whitelist.go [hunk 3]
```diff
@@ -32,92 +36,122 @@ import (
...
+		qResponseType, ok := qResponse.Interface().(codec.ProtoMarshaler)
+		if !ok {
+			panic("can't")
+		}
+		g.QueryReponses = append(g.QueryReponses, qResponseType)
+	}
+}
+
+// RegisterService collects all the query path of a module query service
+func (g *GRPCQueriesInfo) RegisterService(sd *grpc.ServiceDesc, ss interface{}) {
```
[the remark this answers is not in the document]
Comment (hieuvubk) [reply wasmbinding/stargate_whitelist.go:68]: No, we rewrited this func from `query.pb.go` to get all query paths.

File: wasmbinding/queries_list.go [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+	minttypes "github.com/osmosis-labs/osmosis/v12/x/mint/types"
+	poolincentivestypes "github.com/osmosis-labs/osmosis/v12/x/pool-incentives/types"
+	superfluidtypes "github.com/osmosis-labs/osmosis/v12/x/superfluid/types"
+	tokenfactorytypes "github.com/osmosis-labs/osmosis/v12/x/tokenfactory/types"
+	twapquerytypes "github.com/osmosis-labs/osmosis/v12/x/twap/client/queryproto"
+	txfeestypes "github.com/osmosis-labs/osmosis/v12/x/txfees/types"
+)
+
+var (
+	QueriesList = []codec.ProtoMarshaler{
```
[the remark this answers is not in the document]
Comment (hieuvubk) [reply wasmbinding/queries_list.go:25]: I thought it was already in `query_plugin_test.go`

[the remark this answers is not in the document]
Comment (mattverse) [reply wasmbinding/queries_list.go:27]: Just checked, yes it does!

**Review by mattverse — COMMENTED** [review]
Core logic LGTM. Just some few minor comments

File: .github/mergify.yml [hunk 5]
```diff
@@ -12,9 +12,23 @@ queue_rules:
...
+        - check-success=Cosmwasm Lints #contracts
+        - check-success=Test Suite #contracts
+        - check-success=Run golangci-lint #lint
+        - check-success=Run super-linter #lint
+        - check-success=e2e #test
+        - check-success=go #test
+        - check-success=labeler #labeler
+        - check-success~=^osmosis\s[[:alnum:]]+\sfor\s[[:alnum:]]+/g # Build 
+        - check-success=should_run_go_test #test
+        - check-success=test #sim
```
[the remark this answers is not in the document]
Comment (hieuvubk) [reply .github/mergify.yml]: Sr just merging err 😪

File: wasmbinding/stargate_whitelist.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (mattverse) [reply wasmbinding/stargate_whitelist.go:76]: @hieuvubk ^bump on this comment!

**Review by mattverse — APPROVED** [review]
LGTM! Nice work on this! Let's merge this upon one more approval

File: wasmbinding/queries_list.go [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+		&superfluidtypes.AssetMultiplierResponse{},
+		&txfeestypes.QueryFeeTokensResponse{},
+		&txfeestypes.QueryDenomSpotPriceResponse{},
+		&txfeestypes.QueryDenomPoolIdResponse{},
+		&txfeestypes.QueryBaseDenomResponse{},
+		&tokenfactorytypes.QueryParamsResponse{},
+		&tokenfactorytypes.QueryDenomAuthorityMetadataResponse{},
+		&twapv2querytypes.ArithmeticTwapResponse{},
+		&twapv2querytypes.ArithmeticTwapToNowResponse{},
+		&twapquerytypes.ParamsResponse{},
```
Comment (ValarDragon) [inline wasmbinding/queries_list.go:78]: This is far more queries than we had before. Lets please do two different PR's, and leave the query list untouched in teh first one, beyond route removal.

**Comment by hieuvubk** [comment]
@mattverse Sr I left a cmt before but it got lending. RegisterService runs inside `RegisterQueryServer` for each module 
https://github.com/osmosis-labs/osmosis/blob/becbf70519df91690b0ca1ed14a9135f80692a84/x/incentives/types/query.pb.go#L1192-L1194

**Comment by ValarDragon** [comment]
Closing due to staleness. But also, doing a separate query server registration in another file doesn't feel particularly better than the flow rn
